### DISCUSSION TITLE
attachInterrupt( ) - modes ONLOW_WE & ONHIGH_WE, how do they work?

### DISCUSSION THREAD

**Discussion by Algonsi** [espressif/arduino-esp32#6853] [Q&A]
Hello, I have searched everywhere to know how interrupts work with ONLOW_WE and ONHIGH_WE modes, but I have not been able to find anything about it.
Please does anyone know what those two interrupt modes are for?

**Comment by whogarden** [comment]
Do these modes exist?
I found nothing ...

**Reply by Algonsi** [reply]
I guess so, since you can find them in Arduino.h, line 56:

```
//Interrupt Modes
#define RISING    0x01
#define FALLING   0x02
#define CHANGE    0x03
#define ONLOW     0x04
#define ONHIGH    0x05
#define ONLOW_WE  0x0C
#define ONHIGH_WE 0x0D
```

**Reply by platypii** [reply]
> So, in simple words, ONLOW_WE will trigger an interrupt if the input signal level is low (not a falling edge) and wake up the ESP32 from light sleep, if it is currently in light sleep.

https://community.platformio.org/t/attachinterrupt-modes-onlow-we-onhigh-we-how-do-they-work/28289/2

**Reply by Algonsi** [reply]
Thank you so much. Memory failure

**Reply by whogarden** [reply]
Thank you very much **platypii**
It's clear for me now.
This info was not found in  docs.espressif.com/projects/arduino-esp32
